core/env/env_spec_helper.rb

```diff
@@ -0,0 +1,24 @@
+# Reserve variables.
+def reserve_variables(hash)
+  hash.each_pair  do |name, value|
+    fail "Environment variable name #{name} is already in use" if ENV.include?(name)
+  end
+  @reserved_names = hash.keys
+  ENV.update(hash)
+end
```

Comment:
This seems over-engineered for what we need in specs.
And the spec should work, regardless of which environment variables are set.

I think a simple pattern to use is:
```ruby
before :each do
  @foo = ENV["foo"]
end

after :each do
  ENV["foo"] = @foo
end
```
This works even whether the `foo` env var is set or not.